pkg/vmimage/vmimage_test.go

```diff
@@ -0,0 +1,75 @@
+package vmimage
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/openshift/openshift-azure/pkg/util/azureclient"
+	"github.com/openshift/openshift-azure/pkg/util/random"
+	"github.com/openshift/openshift-azure/pkg/util/tls"
+	"github.com/sirupsen/logrus/hooks/test"
+)
+
+func TestGenerateTemplate(t *testing.T) {
+	_, hook := test.NewNullLogger()
+	timestamp := time.Now().UTC().Format("200601021504")
+	image := "rhel7-3.11-" + timestamp
+	ctx := context.Background()
+	domainNameLabel, err := random.LowerCaseAlphaString(20)
+	if err != nil {
+		t.Error(err)
+	}
+	b, err := ioutil.ReadFile("../../secrets/client-key.pem")
```

Comment:
you might have to create dummy keys/certs

Reply:
reuse populate.DummyCertsAndKeys(cs), which has just merged.  See TestHashScaleSetStability for usage.

Reply:
I don't use an `OpenShiftCluster` object here, so just using the `tls.*` calls as `DummyCertsAndKeys()` does